tmp/sirius-script-app-bo-pso-inj.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python-sirius
+
+import time as _time
+import numpy as np
+from epics import PV as _PV
+
+
+class PSO:
+    def __init__(self, dim=5):
+        self.pv_ioc_prefix_SOFB = 'BO-Glob:AP-SOFB:'
+        self.pv_ioc_prefix_PosAng = 'TB-Glob:AP-PosAng'
+        self.pv_orbx = _PV(self.pv_ioc_prefix_SOFB + 'SPassOrbX-Mon')  # um
+        self.pv_orby = _PV(self.pv_ioc_prefix_SOFB + 'SPassOrbY-Mon')  # um
+        self.pv_sum = _PV(self.pv_ioc_prefix_SOFB + 'SPassSum-Mon')  # counts
+        self.pv_dx = _PV(self.pv_ioc_prefix_PosAng + 'DeltaPosX-SP')  # mm
+        self.pv_dxl = _PV(self.pv_ioc_prefix_PosAng + 'DeltaAngX-SP')  # mrad
+        self.pv_dy = _PV(self.pv_ioc_prefix_PosAng + 'DeltaPosY-SP')  # mm
+        self.pv_dyl = _PV(self.pv_ioc_prefix_PosAng + 'DeltaAngY-SP')  # mrad
+        self.pv_resetposang = _PV(
+            self.pv_ioc_prefix_PosAng + 'SetNewRefKick-Cmd')
+        self.pv_kckr = _PV('BO-01D:PM-InjKckr:Kick-SP')
+        _time.sleep(2)
+        self.pv_resetposang.value = 1
+
+        self.x_lim = 3
+        self.y_lim = 3
+        self.xl_lim = 1
+        self.yl_lim = 3
+        self.kckr_lim = 3
+
+        self.p_bpm = 1.0
+        self.p_orb = 1e-7
+
+        self.c_inertia = 0.7984
```

Comment:
if `self.c_inertia`(and possibly other object attributes in this class) is indeed a constant, I recommend you change it to be either a class constant or a module constant. If they are supposed to be values defined by the class user, you might want to make them "hidden" attributes (such as `self._c_inertia`) and add setter and getter methods for these hidden attributes.

Reply:
ok. temporary script...